config/config_test.go

```diff
@@ -189,9 +107,24 @@ func TestGetOptionalParameters(t *testing.T) {
 	t.Run("Live data without sensors", func(t *testing.T) {
 		cfgService := makeCfgService()
 		cfgService.Attributes["use_live_data"] = true
-		cfg, err := NewAttrConfig(cfgService)
+		cfg, err := newAttrConfig(cfgService)
 		test.That(t, err, test.ShouldBeNil)
 		_, _, _, _, _, err = GetOptionalParameters(cfg, "localhost", 1001, 1002, logger)
 		test.That(t, err, test.ShouldBeError, newError("sensors field cannot be empty when use_live_data is set to true"))
 	})
 }
+
+func newAttrConfig(cfg config.Service) (*AttrConfig, error) {
```

Comment:
I kept this as a helper function for the 2 tests that use it within `TestGetOptionalParameters` as it does confirm that the config under test passes the validate function. 

Please let me know if I should do something different.